src/installing-deis/configuring-object-storage.md

```diff
@@ -6,20 +6,124 @@ A variety of Deis components rely on an object storage system to do their work.
+
+As you may know, Google Cloud Storage (GCS) can [interoperate with the S3 API](https://cloud.google.com/storage/docs/interoperability), and, if you choose to use Google Cloud Storage for object storage, you'll have to turn on this interoperability mode.
+
+If you choose to use Google Cloud Storage, set your `DEIS_OUTSIDE_STORAGE` environment variable to `storage.googleapis.com`, and follow [these instructions](https://cloud.google.com/storage/docs/migrating?hl=en_US#keys) to generate an S3 compatible access key ID and access key secret. Store these credentials just as you would if they were AWS S3 or Minio credentials. As mentioned above, we recommend storing these as Kubernetes secrets. See the "Configuring Deis Components" section above for more details and examples.
+
+## [deis/slugbuilder](https://github.com/deis/slugbuilder)
+
+### Environment Variables
+
+The slugbuilder looks for the below environment variables to determine where to download code from and upload slugs to.
+
+- `TAR_URL` - The location of the `.tar` archive (which it will build)
+- `put_url` - The location this component will upload the finished slug to
+
+### Credentials
+
+The slugbuilder reads credentials from the below locations on the filesystem.
+
+- Key: `/var/run/secrets/object/store/access-key-id`
+- Secret `/var/run/secrets/object/store/access-key-secret`
+
+
+## [deis/slugrunner](https://github.com/deis/slugrunner)
+
+### Environment Variables
+
+The slugrunner uses the `SLUG_URL` environment variable to determine where to download the slug (that it will run) from.
+
+### Credentials
+
+The slugrunner reads credentials from the below locations on the filesystem.
+
+- Key: `/var/run/secrets/object/store/access-key-id`
+- Secret: `/var/run/secrets/object/store/access-key-secret`
+
+## [deis/registry](https://github.com/deis/registry)
+
+### Environment Variables
+
+TODO
+
+### Credentials
+
+The registry reads credentials from the below locations on the filesystem.
+
+- Key: `/var/run/secrets/deis/registry/creds/accesskey`
+- Secret: `/var/run/secrets/deis/registry/creds/secretkey`
+
+## [deis/database](https://github.com/deis/postgres)
+
+The database is configured slightly differently from the other components. It looks for an environment variable which it then uses as a key to look up the object storage configuration in a config file.
+
+### Environment Variables
+
+The database looks for the `DATABASE_STORAGE` environment variable to determine what object storage system to use. Valid values are listed below and the [Helm chart for Deis](https://github.com/deis/charts/tree/master/deis-dev) defaults to `minio`:
+
+- filesystem: Store persistent data on ephemeral disk
+- s3: Store persistent data in AWS S3 (configure in S3 section)
+- azure: Store persistent data in Azure's object storage
+- gcs: Store persistent data in Google Cloud Storage
+- minio: Store persistent data on in-cluster Minio server
+
+(TODO: is this correct?)
```

Comment:
The only two supported at this time are `s3` and "anything other than s3", which is minio.